### ISSUE TITLE
easybake crashes because it thinks a string is a step number

### ISSUE THREAD

**Issue by philschatz** [Connexions/cnx-easybake#51]
This commit causes easybake to crash: https://github.com/Connexions/cnx-recipes/commit/2e9f9a048a3b47c33581fccf8ed041e6c43f1e0f (to duplicate, run `./scripts/setup && ./scripts/fetch-book TEAhsphysics && ./scripts/bake-book TEAhsphysics`

I added logging to lookup and got the following (line numbers should be reduced by 2 lines or so because I added logging):

```
 File "/Users/phil/github/cnx/cnx-recipes/venv/bin/cnx-easybake", line 9, in <module>
    load_entry_point('cnx-easybake==0.7.0', 'console_scripts', 'cnx-easybake')()
  File "/Users/phil/github/cnx/cnx-recipes/venv/lib/python2.7/site-packages/cnxeasybake/scripts/main.py", line 81, in main
    args.coverage_file)
  File "/Users/phil/github/cnx/cnx-recipes/venv/lib/python2.7/site-packages/cnxeasybake/scripts/main.py", line 21, in easybake
    oven.bake(html_doc, last_step)
  File "/Users/phil/github/cnx/cnx-recipes/venv/lib/python2.7/site-packages/cnxeasybake/oven.py", line 246, in bake
    strval = u''.join([u'{}'.format(s) for s in value])
  File "/Users/phil/github/cnx/cnx-recipes/venv/lib/python2.7/site-packages/cnxeasybake/oven.py", line 80, in __unicode__
    self.el_id))
  File "/Users/phil/github/cnx/cnx-recipes/venv/lib/python2.7/site-packages/cnxeasybake/oven.py", line 533, in lookup
    if vname in state[step][vtype]:
TypeError: string indices must be integers
```

And the values appear to be:

```
target_id = auto_1d26ed40-152e-4579-9674-f8786a5037a4@2_Example_01_03_03 vtype is strings
step=0 vtype=strings
step=isSolution vtype=strings (causes crash)
```

**Comment by reedstrm** [comment]
Hmm, how did 'step' become "isSolution" ? A slightly shorter reproducer would be nice

**Comment by reedstrm** [comment]
From reading the source, that _should_ be 'vname=isSolution, step=0, vtype=strings'

**Comment by philschatz** [comment]
yeah, I was trying to whittle it down to a small chunk but ran out of time. Thanks for the pointer, that's what I was looking for 😄 : 

> Feels like an order-of-operations issue, since it's a deferred lookup of a string to set an attribute that's probably tripping it.

**Comment by philschatz** [comment]
Fixed in #54!
